pkg/gofr/http/middleware/apikey_auth.go

```diff
@@ -4,11 +4,21 @@ package middleware
 
 import (
 	"net/http"
+
+	"gofr.dev/pkg/gofr/container"
 )
 
+// APIKeyAuthProvider represents a basic authentication provider.
+type APIKeyAuthProvider struct {
+	Keys               []string
+	ValidateFunc       func(apiKey string) bool
+	ValidateFuncWithDB func(c *container.Container, apiKey string) bool
```

Comment:
its not just db, it can be ValidateFuncWithDatasouces, ValidateFuncWithContainer ?

Reply:
Sure changed to `ValidateFuncWithDatasouces `